GVFS/GVFS.Common/NuGetUpgrade/ModernNuGetUpgrader.cs

```diff
@@ -0,0 +1,256 @@
+using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+
+namespace GVFS.Common.NuGetUpgrade
+{
+    public class ModernNuGetUpgrader : NuGetUpgrader
+    {
+        private HttpClient httpClient;
+        private string platform;
+
+        public ModernNuGetUpgrader(
+           string currentVersion,
+           ITracer tracer,
+           PhysicalFileSystem fileSystem,
+           HttpClient httpClient,
+           bool dryRun,
+           bool noVerify,
+           ModernNuGetUpgraderConfig config,
+           string downloadFolder,
+           string platform,
+           ICredentialStore credentialStore)
+           : base(
+               currentVersion,
+               tracer,
+               fileSystem,
+               dryRun,
+               noVerify,
+               config,
+               downloadFolder,
+               credentialStore)
+        {
+            this.httpClient = httpClient;
+            this.platform = platform;
+        }
+
+        public ModernNuGetUpgrader(
+           string currentVersion,
+           ITracer tracer,
+           PhysicalFileSystem fileSystem,
+           HttpClient httpClient,
+           bool dryRun,
+           bool noVerify,
+           ModernNuGetUpgraderConfig config,
+           string platform,
+           NuGetFeed nuGetFeed,
+           ICredentialStore credentialStore)
+           : base(
+               currentVersion,
+               tracer,
+               dryRun,
+               noVerify,
+               fileSystem,
+               config,
+               nuGetFeed,
+               credentialStore)
+        {
+            this.httpClient = httpClient;
+            this.platform = platform;
+        }
+
+        public override bool SupportsAnonymousVersionQuery { get => true; }
+
+        private ModernNuGetUpgraderConfig Config { get => this.nuGetUpgraderConfig as ModernNuGetUpgraderConfig;  }
+        private string OrgInfoServerUrl { get => this.Config.OrgInfoServer; }
+        private string Ring { get => this.Config.UpgradeRing; }
+        private string OrgName
+        {
+            get
+            {
+                if (!AzDevOpsFromNuGetFeedUrlParser.TryParseOrgFromNuGetFeedUrl(this.Config.FeedUrl, out string orgName))

```

Comment:
This will run the regex everytime the code accesses the `OrgName` which is bad IMO.  Can we move this to the constructor so it only happens once?  Or will this be changing and we have to do this each time the `FeedUrl` changes?

Reply:
It could be a lazy property / cached. It is a private property that is only accessed 1 time, so is not actually an issue. It could also be a method (`GetOrgName()`) or something...